packages/cli/src/TestWebhooks.ts

```diff
@@ -274,45 +284,42 @@ export class TestWebhooks implements IWebhookManager {
 		return foundWebhook;
 	}
 
-	async activateWebhook(
-		workflow: Workflow,
-		webhook: IWebhookData,
-		executionMode: WorkflowExecuteMode,
-		activationMode: WorkflowActivateMode,
-	) {
+	async activateWebhook(workflow: Workflow, webhook: IWebhookData) {
 		webhook.path = removeTrailingSlash(webhook.path);
+		webhook.isTest = true;
 
-		const key = this.toWebhookKey(webhook);
+		/**
+		 * Remove additional data from webhook because:
+		 *
+		 * - It is not needed for the test webhook to be executed.
+		 * - It contains circular refs that cannot be be cached.
+		 */
+		const { workflowExecuteAdditionalData: _, ...rest } = webhook;
 
-		webhook.isTest = true;
-		this.registrations[key].webhook = webhook;
+		await this.registrations.updateWebhookProperties(rest as IWebhookData);
 
 		try {
-			await workflow.createWebhookIfNotExists(
-				webhook,
-				NodeExecuteFunctions,
-				executionMode,
-				activationMode,
-			);
+			await workflow.createWebhookIfNotExists(webhook, NodeExecuteFunctions, 'manual', 'manual');
 		} catch (error) {
-			if (this.registrations[key]) delete this.registrations[key];
+			await this.registrations.deregister(webhook);
 
 			throw error;
 		}
 	}
 
-	getActiveWebhook(httpMethod: IHttpRequestMethods, path: string, webhookId?: string) {
-		const key = this.toWebhookKey({ httpMethod, path, webhookId });
-		if (this.registrations[key] === undefined) {
-			return undefined;
-		}
+	async getActiveWebhook(httpMethod: IHttpRequestMethods, path: string, webhookId?: string) {
+		const key = this.registrations.toKey({ httpMethod, path, webhookId });
+
+		const exists = await this.registrations.exists(key);
```

Comment:
Is the `exists` call necessary? Can we get rid of it? We are fetching it right afterwards. We might choose to either return undefined if `get` does not exist or throw an error and handle it. WDYT? This is to prevent another round trip to Redis, that involves also deserializing all the data.